### ISSUE TITLE
💅 Regression: noUnusedPrivateClassMember started reporting false positive in 2.1.2

### ISSUE THREAD

**Issue by vladkrasn** [biomejs/biome#6920]
### Environment information

```bash
Biome playground
```

### Rule name

noUnusedPrivateClassMembers

### Playground link

https://biomejs.dev/playground/?lintRules=noUnusedPrivateClassMembers&code=YwBsAGEAcwBzACAAQwBvAHUAbgB0AGUAcgAgAHsACgAgACAAcAByAGkAdgBhAHQAZQAgAGMAbwB1AG4AdABlAHIAOgAgAG4AdQBtAGIAZQByAAoAIAAgAGEAcwB5AG4AYwAgAGMAbwB1AG4AdAAoACkAIAB7AAoAIAAgACAAIAB0AGgAaQBzAC4AYwBvAHUAbgB0AGUAcgAgAD0AIAAxAAoAIAAgACAAIABjAG8AbgBzAHQAIABjAG8AdQBuAHQAZQByAFMAdAByAGkAbgBnACAAPQAgAGAAJAB7AHQAaABpAHMALgBjAG8AdQBuAHQAZQByACsAKwB9AGAACgAgACAAfQAKAH0A

### Expected result

`this.counter` is used actually! There should be no error

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
cc @vladimir-ivanov

**Comment by vladkrasn** [comment]
@vladimir-ivanov , the playground example I provided still shows errors.

**Comment by mrazauskas** [comment]
Hm.. Seems like Biome 2.1.3 included the patch fixing this issue, but it does not work or me.

A minimal example like:

```ts
export class Counter3 {
  private counter = 1;

  count() {
    return this.counter++;
  }
}
```

also fails in the playground:

https://biomejs.dev/playground/?lintRules=noUnusedPrivateClassMembers&code=ZQB4AHAAbwByAHQAIABjAGwAYQBzAHMAIABDAG8AdQBuAHQAZQByADMAIAB7AAoAIAAgAHAAcgBpAHYAYQB0AGUAIABjAG8AdQBuAHQAZQByACAAPQAgADEAOwAKAAoAIAAgAGMAbwB1AG4AdAAoACkAIAB7AAoAIAAgACAAIAByAGUAdAB1AHIAbgAgAHQAaABpAHMALgBjAG8AdQBuAHQAZQByACsAKwA7AAoAIAAgAH0ACgB9AA%3D%3D

Or did I misunderstood something?
